### PR TITLE
Easy access to Zephir function name in AbstractOptimizer

### PR DESCRIPTION
**Pull request by flip111**

### CODE REVIEW

**Comment by sergeyklay** [comment]
@flip111 What issue does it solve?

File: Library/Optimizers/OptimizerAbstract.php [hunk 1]
```diff
@@ -21,12 +21,20 @@
...
 use Zephir\Call;
 use Zephir\CompilationContext;
+use Zephir\Utils;
 
 /**
  * Class OptimizerAbstract
  */
 abstract class OptimizerAbstract
 {
+	protected $functionName;
```
Comment (sergeyklay) [inline Library/Optimizers/OptimizerAbstract.php:31]: Spaces must be used to indent lines; tabs are not allowed

File: Library/Optimizers/OptimizerAbstract.php [hunk 2]
```diff
@@ -21,12 +21,20 @@
...
+
+	public function __construct() {
```
Comment (sergeyklay) [inline Library/Optimizers/OptimizerAbstract.php:33]: Opening brace should be on a new line

**Review by sergeyklay — CHANGES_REQUESTED** [review]

**Comment by flip111** [comment]
I forgot the specific case, but in case you have several optimizers calling the same function you can now pass information to C which function it was.

**Pull request closed without merging**